sound/soc/sof/ipc4-topology.c

```diff
@@ -1749,6 +1750,15 @@ static int sof_ipc4_prepare_process_module(struct snd_sof_widget *swidget,
 	}
 
 	memcpy(cfg, data, control_data->data->size);
+
+	/* Update pipeline params from output format */
+	out_fmt = sof_ipc4_get_widget_pin_format(swidget, SOF_PIN_TYPE_SOURCE, 0);
+	/* Currently, only smart amp module needs to update channels */
+	hw_param_interval(pipeline_params, SNDRV_PCM_HW_PARAM_CHANNELS)->min =
+		SOF_IPC4_AUDIO_FORMAT_CFG_CHANNELS_COUNT(out_fmt->fmt_cfg);
+	hw_param_interval(pipeline_params, SNDRV_PCM_HW_PARAM_CHANNELS)->max =
+		SOF_IPC4_AUDIO_FORMAT_CFG_CHANNELS_COUNT(out_fmt->fmt_cfg);
```

Comment:
I don't understand what this is about.

a) does a pipeline have format parameters? If not, then the commit message is just wrong.
b) can avoid the repetition of hw_param_interval(pipeline_params, SNDRV_PCM_HW_PARAM_CHANNELS)-> and use a local variable to make the code more readable?